### DISCUSSION TITLE
Is it possible to add individual project pages?

### DISCUSSION THREAD

**Discussion by jxchen01** [greenelab/lab-website-template#163] [General] [answered]
Hello, I am trying a creat a list of posts or cards under the "Research" page, not about publications. For example, I have three different projects in my group. I want to create three cards or three posts on the page of "Research", so that when people click on one card / post, they will be directed to a separated page about that specific project. Then, on that page, I can add more details of this project, add all publications related to this project, add funding info, etc.. 

Currently, if I implemented these projects as cards, then they can only be linked to an external page, like a github project page, or some public url. I am thinking something like posts, but certainly different from the current blog posts. I don't know how exactly I could do this.

Any suggestions?

**Comment by vincerubinetti** [comment, marked as answer]
Hi, good question. I think this is something many people could potentially want.

First, check out these pages of the docs for some background information:
https://greene-lab.gitbook.io/lab-website-template-docs/advanced/custom-components
https://greene-lab.gitbook.io/lab-website-template-docs/advanced/data-and-collections

Since you want separate pages for each item, you'd want a "collection" (instead of "data").

Here's basically what you can do:

1. Create a collection folder called `_projects` (or whatever you want to name it).
2. In your `_config.yaml`, set `output: true` for your new `projects` collection so Jekyll generates a separate page for each item.
3. If you do use "projects" for the name, delete the built-in `/_data/projects.yaml` to avoid name conflict.
4. Create `.md` files in `_projects` for each project you have.
5. In each `.md` file, create a [front matter](https://assemble.io/docs/YAML-front-matter.html) with the same params as the [card component](https://greene-lab.gitbook.io/lab-website-template-docs/basics/components/card)
6. In each `.md` file, below the front matter, add your detailed content as you said, e.g. publications, funding info, etc. You'd want to add all the publications to `sources.yaml` and follow the usual citation process, then just [lookup the ones relevant to each project by id](https://greene-lab.gitbook.io/lab-website-template-docs/basics/components/citation).
7. Finally, on your main `/projects/index.md` page, use the [list component](https://greene-lab.gitbook.io/lab-website-template-docs/basics/components/list) to loop through the `_projects` items, and display each front-matter with the card component, e.g. `{% include list.html data="projects" component="card" %}`.

I hope this helps.

I may add this as an example in the docs.

**Reply by jxchen01** [reply]
Thanks so much for your detailed instruction. I have it almost working. 95% done. The last thing is that the specific citation cannot be displayed properly on the project page. Here is what I used: https://github.com/MMV-Lab/mmv-lab.github.io/blob/main/_projects/cell-tracking.md?plain=1#L25.  Any idea what could be wrong?

**Reply by vincerubinetti** [reply]
`doi:10.1016/j.csbj.2020.05.017` is not in your `sources.yaml` or `citations.yaml`. You have to add it to your `sources.yaml` then run the [cite process](https://greene-lab.gitbook.io/lab-website-template-docs/basics/citations).

**Reply by jxchen01** [reply]
hmm.. sorry for my stupid mistake. But, it still cannot display properly after fixing the ID. https://github.com/MMV-Lab/mmv-lab.github.io/blob/main/_projects/cell-tracking.md?plain=1#L25. This time, I have checked the ID exists in `citations.yaml`: https://github.com/MMV-Lab/mmv-lab.github.io/blob/main/_data/citations.yaml#L187

**Reply by vincerubinetti** [reply]
It looks like you're using a pretty old version of the template where the parameter was named `id` instead of `lookup` in the citation component. You can see this in `/_includes/citation.html`.

It was renamed to `lookup` because in later versions it can be looked up by the title as well as the id.

**Reply by jxchen01** [reply]
ok, after changing it to {% include citation.html id="doi:10.1007/978-3-319-24574-4_34" style="rich" %}, everything works fine now. 

I also did several attempts to merge all upstream updates, but messed up at the end. So, I have to stick with the olde version. Is there any suggestion on how to best sync with the template repo?

**Reply by vincerubinetti** [reply]
There is a dedicated docs page for updating here:
https://greene-lab.gitbook.io/lab-website-template-docs/advanced/update-your-template

Which also links to and involves this page which explains which files/folders belong to the template, and which belong to the user:
https://greene-lab.gitbook.io/lab-website-template-docs/basics/repo-structure

You should also take a look at the breaking changes here:
https://github.com/greenelab/lab-website-template/blob/main/CHANGELOG.md

Unfortunately, since this is a template (where you get all the under the hood code) and not a cleanly installable package (with a small surface area API), it's simply difficult to update. Everything is tied together. I've tried to separate things as cleanly as possible, but Jekyll is very limiting here.

The easiest way to help you would probably be if you could open a PR against your main branch with the changes, and then I can look at the diff and see what might be broken.

**Reply by jxchen01** [reply]
thanks! this is soooooo helpful. maybe I can close this thread for now?
